### PR TITLE
Explain loss function and optimization 

### LINKED ISSUE #251: Add more explanation on how loss function and optimization works
It would be nice to add a bit more theory on loss functions and optimization. We often get questions about it and explain along the way by improvising, but it would be nice if there is some material about it with pictures. Something like: 
image
and
image

This would fit in https://carpentries-incubator.github.io/deep-learning-intro/02-keras/index.html#5-choose-a-loss-function-and-optimizer

### PR DESCRIPTION
**Pull request by svenvanderburg**
Fixes #251 #106

### CODE REVIEW

File: episodes/1-introduction.Rmd [hunk 1]
```diff
@@ -104,6 +105,48 @@ This image is from the paper ["An Efficient Pedestrian Detection Method Based on
...
+|                | **MSE:**           | 3.5               |
+
+::::
+:::
+
+So, a loss function quantifies the total error of the model.
+The process of adjusting the weights in such a way as to minimize the loss function is called 'optimization'.
+We will dive further into how optimization works in episode 3,
+for now it is enough to understand that during training the weights in the network are adjusted so that the loss decreases through the process of optimization.
+Ultimately resulting in a low loss, and thus predictions that are close to the expected values.
```
Comment (cpranav93) [inline episodes/1-introduction.Rmd:149]:
```suggestion
This ultimately results in a low loss, and this, generally, implies predictions that are closer to the expected values.
```

File: episodes/1-introduction.Rmd [hunk 2]
```diff
@@ -104,6 +105,48 @@ This image is from the paper ["An Efficient Pedestrian Detection Method Based on
...
+| 3              | 2                  | 1                 |
+|                | **MSE:**           | 3.5               |
+
+::::
+:::
+
+So, a loss function quantifies the total error of the model.
+The process of adjusting the weights in such a way as to minimize the loss function is called 'optimization'.
+We will dive further into how optimization works in episode 3,
+for now it is enough to understand that during training the weights in the network are adjusted so that the loss decreases through the process of optimization.
```
Comment (cpranav93) [inline episodes/1-introduction.Rmd:148]:
```suggestion
We will dive further into how optimization works in episode 3.
For now, it is enough to understand that during training the weights in the network are adjusted so that the loss decreases through the process of optimization.
```

File: episodes/3-monitor-the-model.Rmd [hunk 3]
```diff
@@ -254,6 +256,75 @@ Non-trainable params: 0
...
+- B. The weights are adjusted so that we move in the direction of the gradient, so down the slope of the loss function
+- C. The weights are adjusted so that we move in the direction of the negative gradient, so up the slope of the loss function
+- D. The weights are adjusted so that we move in the direction of the negative gradient, so down the slope of the loss function
+
+### 3. When the batch size is increased:
+(multiple answers might apply)
+
+- A. The number of samples in an epoch also increases
+- B. The number of batches in an epoch goes down
+- C. The training progress is less jumpy, because more samples are consulted in each update step (one batch).
```
Comment (cpranav93) [inline episodes/3-monitor-the-model.Rmd:306]: I think the statement should be:
```suggestion
- C. The training progress is morejumpy, because more samples are consulted in each update step (one batch).
```

This makes sense and also what it states in the solution

**Review by cpranav93 — APPROVED** [review]
Added some comments to address but otherwise great work @svenvanderburg! I think it adds some much needed material to the lesson. Lets give it a go in the next workshop!

File: episodes/3-monitor-the-model.Rmd (hunk 3 above)
Comment (svenvanderburg) [reply episodes/3-monitor-the-model.Rmd:306]: Well spotted! Thanks :)

**Comment by svenvanderburg** [comment]
Thanks for your comments @cpranav93 🙏
